tests/qinfo/test_entropies.py

```diff
@@ -953,3 +953,399 @@ def circuit(params):
+        ]
+
+        param0, param1 = tf.Variable(param[0]), tf.Variable(param[1])
+        with tf.GradientTape() as tape:
+            out = qml.qinfo.relative_entropy(circuit1, circuit2, [0], [1])((param0,), (param1,))
+
+        actual = tape.gradient(out, [param0, param1])
+
+        assert np.allclose(actual, expected, atol=1e-5)
+
+    @pytest.mark.torch
+    @pytest.mark.parametrize("param", grad_params)
+    def test_qnode_grad_torch(self, param):
+        """Test that the gradient of relative entropy works for QNodes
+        with the Torch interface"""
+        import torch
+
+        dev = qml.device("default.qubit", wires=2)
+
+        @qml.qnode(dev, interface="torch", diff_method="backprop")
+        def circuit1(param):
+            qml.RY(param, wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+        @qml.qnode(dev, interface="torch", diff_method="backprop")
+        def circuit2(param):
+            qml.RY(param, wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+            first_expected = (
+                np.sin(param[0] / 2)
+                * np.cos(param[0] / 2)
+                * (np.log(np.tan(param[0] / 2) ** 2) - np.log(np.tan(param[1] / 2) ** 2))
+            )
+
+        expected = [
+            np.sin(param[0] / 2)
+            * np.cos(param[0] / 2)
+            * (np.log(np.tan(param[0] / 2) ** 2) - np.log(np.tan(param[1] / 2) ** 2)),
+            np.cos(param[0] / 2) ** 2 * np.tan(param[1] / 2)
+            - np.sin(param[0] / 2) ** 2 / np.tan(param[1] / 2),
+        ]
+
+        param0 = torch.tensor(param[0], requires_grad=True)
+        param1 = torch.tensor(param[1], requires_grad=True)
+        out = qml.qinfo.relative_entropy(circuit1, circuit2, [0], [1])((param0,), (param1,))
+        out.backward()
+
+        actual = [param0.grad, param1.grad]
+
+        assert np.allclose(actual, expected, atol=1e-8)
+
+    @pytest.mark.all_interfaces
+    @pytest.mark.parametrize("device", ["default.qubit", "default.mixed", "lightning.qubit"])
+    @pytest.mark.parametrize("interface", ["autograd", "jax", "tensorflow", "torch"])
+    def test_num_wires_mismatch(self, device, interface):
+        """Test that an error is raised when the number of wires in the
+        two QNodes are different"""
+        dev = qml.device(device, wires=2)
+
+        @qml.qnode(dev, interface=interface)
+        def circuit1(param):
+            qml.RY(param, wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+        @qml.qnode(dev, interface=interface)
+        def circuit2(param):
+            qml.RY(param, wires=0)
+            qml.CNOT(wires=[0, 1])
+            return qml.state()
+
+        msg = "The two states must have the same number of wires"
+        with pytest.raises(qml.QuantumFunctionError, match=msg):
+            rel_ent_circuit = qml.qinfo.relative_entropy(circuit1, circuit2, [0], [0, 1])
+
+    @pytest.mark.autograd
```

Comment:
I think the three last tests do not need to be marked with autograd as we don't use any gradient.